src/averageAges.js

```diff
@@ -36,8 +45,23 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+function calculateWomenAverageAge(people, withChildren = false) {
+  if (withChildren === false) {
+    const womenArray = people.filter(woman => woman.sex === 'f');
+    const yearsSum = womenArray.reduce((years, woman) =>
+      years + (woman.died - woman.born), 0);
+
+    return yearsSum / womenArray.length;
+  } else {
+    const mothersArray = people.map(woman => woman.mother);
+    const womenWithChildrenArray = people.filter(woman =>
```

Comment:
Instead of filtering for 'woman.sex === 'f'' twice (line 36 and 44), you can filter once for all women and then filter for women with children.